File: app/src/main/java/com/gpetuhov/samplemockito/SampleMockitoApp.java
package com.gpetuhov.samplemockito;

import android.app.Application;

import com.gpetuhov.samplemockito.dagger.components.DaggerDataComponent;
import com.gpetuhov.samplemockito.dagger.components.DataComponent;

public class SampleMockitoApp extends Application {

    private static DataComponent dataComponent;
    private static SampleMockitoApp sampleMockitoApp;

    @Override
    public void onCreate() {
        super.onCreate();

        sampleMockitoApp = this;

        dataComponent = DaggerDataComponent.builder()
                .build();
    }

    public static DataComponent getDataComponent() {
        return dataComponent;
    }

    public static SampleMockitoApp getApplication() {
        return sampleMockitoApp;
    }
}

File: app/src/main/java/com/gpetuhov/samplemockito/SampleMockitoApp.kt
package com.gpetuhov.samplemockito

import android.app.Application

import com.gpetuhov.samplemockito.dagger.components.DaggerDataComponent
import com.gpetuhov.samplemockito.dagger.components.DataComponent

class SampleMockitoApp : Application() {

    companion object {
        lateinit var dataComponent: DataComponent
        lateinit var application: SampleMockitoApp
    }

    override fun onCreate() {
        super.onCreate()
        application = this
        dataComponent = DaggerDataComponent.builder().build()
    }
}
